apps/client/src/pages/onBoarding/components/TimeSelectStep.tsx

```diff
@@ -1,23 +1,68 @@
+import { useEffect, useRef, useState } from 'react';
 import { STEP } from '@pages/onBoarding/types';
 import { Icon } from '@pinback/design-system/icons';
-import { useState } from 'react';
+import OnBoardingTimePicker from '@pages/onBoarding/components/OnboardingTimePicker';
+import OnboardingNavButton from '@pages/onBoarding/components/OnboardingNavButton';
+import { getFormattedSelectedTime } from '@pages/onBoarding/utils/formatSelectedTime';
 
 interface TimeSelectStepProps {
   setStep: (step: string) => void;
 }
 
+const TIME_PRESETS = [
+  { key: 'morning', label: '아침형 치삐', time: '오전 9시' },
+  { key: 'evening', label: '저녁형 치삐', time: '오후 8시' },
+];
+
 const TimeSelectStep = ({ setStep }: TimeSelectStepProps) => {
-  const [selectedTime, setSelectedTime] = useState<string | null>(null);
+  const [selectedTime, setSelectedTime] = useState<string | null>('morning');
+  const [selectedCustomTime, setSelectedCustomTime] = useState<string | null>(
+    null
+  );
   const [isPopupOpen, setIsPopupOpen] = useState(false);
+  const popupRef = useRef<HTMLDivElement>(null);
 
-  const handlePopupToggle = () => {
-    setIsPopupOpen((prev) => !prev);
+  const handleSave = (time: {
+    hour: string;
+    minute: string;
+    meridiem: string;
+  }) => {
+    const formattedTime = `${time.meridiem} ${time.hour}시${time.minute !== '0' ? ` ${time.minute}분` : ''}`;
+    setSelectedCustomTime(formattedTime);
+    setSelectedTime('custom');
+    setIsPopupOpen(false);
   };
 
-  const handleTimeSelect = (time: string) => {
-    setSelectedTime((prev) => (prev === time ? null : time));
+  const handleTimeBoxClick = (key: string) => {
+    if (key === 'custom') {
+      setSelectedTime('custom');
+      setIsPopupOpen((prev) => !prev);
+    } else {
+      setSelectedTime((prev) => (prev === key ? null : key));
+      setIsPopupOpen(false);
+    }
   };
 
+  // 외부 클릭 시 팝업 닫기
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        popupRef.current &&
+        !popupRef.current.contains(event.target as Node)
+      ) {
+        setIsPopupOpen(false);
+      }
+    };
+
+    if (isPopupOpen) {
+      document.addEventListener('mousedown', handleClickOutside);
+    }
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [isPopupOpen]);
+
```

Comment:
코드 방향성은 좋지만 `handleClickOutside`를 `useEffect`안에서 정의해주고 있어서 팝업 열리고 닫힐 때마다 해당 함수가 계속 재정의 될 것 같아요! 그래서 해당 함수는 밖에서 정의하고 쓰는 것은 어떨까요?? 

그리고 더 나아가서 요소 밖을 클릭해서 닫는 로직은 다른 곳에서도 충분히 재사용 될 수 있는 부분이라 custom hook으로 빼서 재사용 할 수 있도록 하면 어떨까요? (shared) 예시로 아래처럼 작성하는 방법 느낌! 아래 코드가 절대 정답은 아니고 custom hook으로 분리하는 방법으로 재사용성을 늘리면 좋을 것 같아요!! 👍 

```suggestion
import { useEffect, useCallback } from 'react';

const useOutsideClick = (
  isOpen: boolean,
  ref: React.RefObject<HTMLElement>,
  onClose: () => void
) => {
  const handleClickOutside = useCallback(
    (event: MouseEvent) => {
      if (ref.current && !ref.current.contains(event.target as Node)) {
        onClose();
      }
    },
    [ref, onClose]
  );

  useEffect(() => {
    if (!isOpen) return;

    document.addEventListener('mousedown', handleClickOutside);
    return () => {
      document.removeEventListener('mousedown', handleClickOutside);
    };
  }, [isOpen, handleClickOutside]);
};
```